### ISSUE TITLE
MPC randomly lags/stutters during video playback

### ISSUE THREAD

**Issue by mandosrex** [clsid2/mpc-hc#1809]
So, every once in a while, without doing anything and with the default settings the player comes with (I haven't touched anything), the player will begin lagging/stuttering during video playback for no apparent reason. The issue happens often between scene transitions (so if the video changes from one scene with a certain background/thing happening, to a completely new scene with a different background), and it goes away after the scene changes, but also during scenes. It also seems to have a higher chance of happening if I scroll through the video with the arrow keys. The lag is only visual, audio is not affected.

Here's a video of the issue happening. The issue happens at the following timestamps: 0:29, 0:51, 1:24, 1:43, 3:11, 3:27, 3:53, 4:03, 4:14, 4:28, 4:50
https://youtu.be/_z3GuU7j2PY
(Note: even though the video file has 4K in its name at the start of the video, the video was converted to 1080p 60FPS. This is irrelevant anyway as all videos I watch on my PC are 1080p and it happens with all of them)

I have an RTX 2060 and a Ryzen 5 2600x and MPC is running from a SSD, so I see no obvious reason for this issue to happen. It almost looks like a hardware acceleration issue but again, it doesn't happen in other things (other players, browser/youtube, games) and I haven't changed any settings. Here's a screenshot of the hardware acceleration menu as well: https://i.imgur.com/WapXugX.png

I also don't think it ever happened with the original defunct MPC from K-Lite Codec Pack, which I used until a few years ago (ran on the same PC). I had hoped updates would eventually fix this issue for me with this MPC but that has yet to happen. I also used VLC recently and it doesn't happen with it either. And yes, I have reset it and reinstalled it as well.

I have used MPC for more than a decade and I really don't want to switch to something else (VLC sucks). Please help.

**Comment by clsid2** [comment]
K-Lite uses the latest version of MPC-HC.

Which video renderer are you using? Have you tried other ones?
Does it only happen with 60fps video, or also with 24/30fps content?
Test with hardware acceleration disabled.

**Comment by mandosrex** [comment]
Oh. I remember reading something about the old MPC being abandoned by its original dev(s) and then 2 new continuations being made by 2 different devs, this one and another version. Does K-Lite now use this version or am I just completely misinformed?

Anyway, tell me where to look for video renderer.

**Comment by clsid2** [comment]
It uses this version.

Options >Playback > Output

**Comment by mandosrex** [comment]
https://i.imgur.com/3hGNTFj.png

**Comment by clsid2** [comment]
Try this:
Menu > View > Renderer settings > VSync > enable Vsync

**Comment by mandosrex** [comment]
Ok, I'm back. I thought of dropping the issue and try to live with it but something interesting and unexpected happened. I stumbled upon a piece of video footage (the end credits of an anime episode) which **always** has the issue. I don't understand why.
Anyway, Vsync was already on and I tried it with it off as well, issue still happens. Let me know what else to try with this section I found to always have the issue, but I feel like this is something unfixable for whatever reason.

**Comment by clsid2** [comment]
Give a (torrent) link to that anime episode. Then I can test it.

**Comment by mandosrex** [comment]
Oh. Well good thing it's something mainstream, lol.
https://nyaa.si/view/1552926
Download episode 5 from this torrent and skip to the end credits with the rap battle. Framerate completely dies for me when the blonde guy starts singing. If you want I can record it again but it's exactly like in the previous video, it goes from 30 FPS or whatever to like 5.
Though I doubt you'll have the issue. But thanks.

**Comment by clsid2** [comment]
It is caused by the animated subtitles. You can toggle them on/off by pressing W.

I need to investigate why this particular subtitle is rendering so slow.

**Comment by mandosrex** [comment]
It is! Huh. I didn't even consider the subs because... Well as you can see this is the dual audio version and it also has dubbed audio (but the credits song is not dubbed). Interesting.
However, in the nature video in my initial post about this issue, there were no subtitles and the issue still happened. But yes, it seems subtitles might be a factor, and the fancier they are the worse the issue is for me.
Is the issue GPU related? But I have not seen issues in other things and the GPU is like two years old.

**Comment by clsid2** [comment]
No, subtitle rendering is done on the CPU. Certain animated effects and blurs are very performance consuming. This one is extreme though.

I can't say much about the nature video without having that file for testing as well.

**Comment by mandosrex** [comment]
Well I simply converted this video https://www.youtube.com/watch?v=LXb3EKWsInQ to 1080p with this website https://en.y2mate.is/38/youtube-to-mp4.html and downloaded it and played it with MPC.
I thought that going with as high of a quality video as possible with 60 FPS will make the lag more noticeable.

**Comment by mandosrex** [comment]
If the subtitle lag is from the CPU then... And before I ask know that I have no based knowledge about these things, but is there like a "multicore rendering" option in MPC like there is in games that I could turn on or something, or that maybe should be turned on by default and maybe it's not for me?

**Comment by mandosrex** [comment]
Should I install K-Lite again? Do I need codecs? Since I read that the MPC from K-Lite stopped being updated, I ditched K-Lite and only used MPC. The first one I found was actually the other MPC I mentioned before, the one with the more modern-ish interface/buttons (more aero?), but then I found yours and have only used it since, no extra codecs.

**Comment by clsid2** [comment]
You read wrong. K-Lite has the newest stuff.

There is no magic setting to improve subtitle performance.

**Comment by clsid2** [comment]
I looked at this subtitle and I am not going to bother trying to improve it. It is super complex. These anime idiots keep making these bullshit scripts that are unnecessarily complex.  I just don't fucking care anymore as it seems it is just intentional by these CP loving faggots.

**Comment by mandosrex** [comment]
Ok, well, I understand the frustration and it's fine. I just wanted to share the latest discoveries I found in case they might help and then I'm out:
I talked with a friend and he told me to try PotPlayer. I did and in that same credits section from that episode, something different happened: instead of the entire video lagging as it did with MPC, only the subtitles lag now, and only when they are longer. If they only have a few words they appear fine.
But I went back and tried installing the latest K-Lite as well and no difference with it. MPC-HC still lags the whole video.

**However**, I rediscovered that the other MPC I talked about is Black Edition. So I gave that a try as a standalone (uninstalled K-Lite) as well, and interestingly, the same issue as with PotPlayer happened, only the subs lag now, not the entire video.

So in short:
- this MPC-HC: whole video lags
- MPC Black Edition and PotPlayer: only subs lag

Maybe this helps. Thanks for your time.

**Comment by clsid2** [comment]
Options > Subtitles > Buffer = 10
And also enable "allow dropping subpictures if queue is running late"

That can help reduce video stutter, but some subs can get skipped.

**Comment by donnaken15** [comment]
> Here's a video of the issue happening. The issue happens at the following timestamps: 0:29, 0:51, 1:24, 1:43, 3:11, 3:27, 3:53, 4:03, 4:14, 4:28, 4:50 https://youtu.be/_z3GuU7j2PY (Note: even though the video file has 4K in its name at the start of the video, the video was converted to 1080p 60FPS. This is irrelevant anyway as all videos I watch on my PC are 1080p and it happens with all of them)

this has been happening to me too
I just updated to 2.0.0.116 (going from a March 2022 update) and it stuttered a video for a moment again (and then apparently stopped when playing the same video for like 20 more minutes)
I also have the stuttering on recording (same way as what OP showed), and happen on playing a video that has next to no visual lag
sometimes reseeking the video appeared to make it go away, but that started to help less lately

edit: it just started doing it again
